src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/resource_allocation_manager_impl.cc

```diff
@@ -246,21 +248,21 @@ ResourceAllocationManagerImpl::CreateOnRCStatusNotificationToMobile(
       mobile_apis::FunctionID::OnRCStatusID, app_id);
   auto& msg_params = (*to_mobile_msg)[application_manager::strings::msg_params];
   ConstructOnRCStatusNotificationParams(
-      msg_params, allocated_resources_, all_supported_modules());
+      msg_params, allocated_resources_, all_supported_modules(), app_id);
   return to_mobile_msg;
 }
 
 smart_objects::SmartObjectSPtr
 ResourceAllocationManagerImpl::CreateOnRCStatusNotificationToHmi(
-        const uint32_t app_id) {
+    const uint32_t hmi_app_id, const uint32_t mobile_app_id) {
```

Comment:
@LitvinenkoIra What if you pass by mistake not matching IDs? There are functions: 
`ApplicationSharedPtr application_by_hmi_app(uint32_t hmi_app_id)` - you will get application
And application has `uint32_t app_id() const ` - so you will get correct app_id.